CMakeLists.txt

```diff
@@ -50,37 +129,64 @@ LIST(APPEND BOOST_COMPONENTS thread
                              program_options
                              chrono
                              unit_test_framework
-                             context)
+                             context
+                             coroutine)
 SET( Boost_USE_STATIC_LIBS ON CACHE STRING "ON or OFF" )
 
-IF( WIN32 )
-  SET(BOOST_ROOT $ENV{BOOST_ROOT})
+IF(WIN32)
+   if($ENV{BOOST_ROOT})
+       SET(BOOST_ROOT $ENV{BOOST_ROOT})
+   endif($ENV{BOOST_ROOT})
   set(Boost_USE_MULTITHREADED ON)
   set(BOOST_ALL_DYN_LINK OFF) # force dynamic linking for all libraries
+  add_definitions("-DCURL_STATICLIB")
+  list(APPEND PLATFORM_SPECIFIC_LIBS ws2_32 crypt32 mswsock userenv )
 ENDIF(WIN32)
 
 FIND_PACKAGE(Boost 1.57 REQUIRED COMPONENTS ${BOOST_COMPONENTS})
-# For Boost 1.53 on windows, coroutine was not in BOOST_LIBRARYDIR and do not need it to build,  but if boost versin >= 1.54, find coroutine otherwise will cause link errors
-IF(NOT "${Boost_VERSION}" MATCHES "1.53(.*)")
-   SET(BOOST_LIBRARIES_TEMP ${Boost_LIBRARIES})
-   FIND_PACKAGE(Boost 1.54 REQUIRED COMPONENTS coroutine)
-   LIST(APPEND BOOST_COMPONENTS coroutine)
-   SET(Boost_LIBRARIES ${BOOST_LIBRARIES_TEMP} ${Boost_LIBRARIES})
-ENDIF()
+
+FIND_PACKAGE(Python 3.6 COMPONENTS Interpreter)
```

Comment:
Would older versions work as well?

Reply:
I have tested it only on Python 3.6+, but I guess even older Python 3 would work.
I would have to make sure, if you are interested in lowering the required version.

Reply:
AFAICS Ubuntu 16.04 LTS ships with Python 2.7 and Python 3.5 by default. If 3 is required, I'd recommend that we check whether 3.5 would work.

Reply:
Just tested on an older system with `cmake-3.5` and both `python-2` and `python-3.4`. `/usr/bin/python` is `python-2` and `/usr/bin/python3` is `python-3`. I changed this line to require python 3.4, but the build fails with two errors:
```
CMake Error at CMakeModules/FindPython/Support.cmake:8 (cmake_policy):
  An attempt was made to set the policy version of CMake to "3.7" which is
  greater than this version of CMake.  This is not allowed because the
  greater version may have new policies not known to this CMake.  You may
  need a newer CMake version to build this project.
...
CMake Error at CMakeModules/FindPython/Support.cmake:226 (if):
  if given arguments:

    "_Python_VERSION" "VERSION_GREATER_EQUAL" "Python_FIND_VERSION"

  Unknown arguments specified
```

Reply:
From my tests, the script is still usable with Python 3.3, and with a small two line modification, also with Python 3.2.

The CMake error comes from the FindPython CMake scripts taken from a recent CMake distribution.

I will try to make it usable with older CMake versions ( but for sure this won't support CMake 2 ), or the second option is, to skip Python checking in older CMake versions. In these cases, this would fall back to using the C++ build helpers, without the need for Python.

Reply:
Another update,

The **first CMake** that is usable with the most up to date version of OpenSSL ( 1.0.2r ) **is 3.1.2**.

Any version older than that will fail, with problems parsing version information from OpenSSL headers.

I have reworked the `FindPython` and `FindBoost` routines to be more future-proof.

Updates are coming soon.